rskj-core/src/main/java/co/rsk/peg/Federation.java

```diff
@@ -52,6 +55,10 @@ public Federation(List<BtcECKey> publicKeys, Instant creationTime, long creation
         // Immutability provides protection unless unwanted modification, thus making the Federation instance
         // effectively immutable
         this.publicKeys = Collections.unmodifiableList(publicKeys.stream().sorted(BtcECKey.PUBKEY_COMPARATOR).collect(Collectors.toList()));
+        this.rskPublicKeys = publicKeys.stream()
```

Comment:
We need to sort these keys too (or either map from `this.publicKeys`) so that we get the same ordering (and therefore index correspondence). A bigger refactor should probably be scheduled in the future so that we reify the concept of a `FederationMember` and its keys.

Oh, btw, I'd personally just map from `this.publicKeys` so that we keep it DRY.

Reply:
Also, we should wrap the member with `Collections.unmodifiableList` so that we keep instances effectively `final`.